src/App.tsx

```diff
@@ -1,14 +1,46 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { SortType } from './types/SortType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selected, setSelected] = useState<Todo | undefined>();
+  const [query, setQuery] = useState('');
+  const [sortQuery, setSortQuery] = useState<SortType>(SortType.ALL);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then(e => {
+        return e.filter(todo =>
+          todo.title.toLowerCase().includes(query.toLowerCase()),
+        );
+      })
+      .then(e => {
+        return e.filter(todo => {
+          switch (sortQuery) {
+            case SortType.ACTIVE:
+              return !todo.completed;
+            case SortType.COMPLETED:
+              return todo.completed;
+            default:
+              return todo;
+          }
+        });
+      })
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, [sortQuery, query]);
```

Comment:
You don't need to fetch data on query change. You can fetch data only once, and then just filter data in your state